packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -213,6 +214,13 @@ export const AutocompleteInput = <
         ...rest,
     });
 
+    const finalOnBlur = useCallback((): void => {
+        if (clearOnBlur) {
+            setFilterValue('');
```

Comment:
I've tested it with the storybook, and it feels weird. Because if I blur the field without selecting anything, the field appears empty *but the actual value hasn't changed*, so the SaveButton isn't enabled. 

We should set the filter value to the current value in this case.